mongoengine/context_managers.py

```diff
@@ -257,3 +258,13 @@ def set_write_concern(collection, write_concerns):
     combined_concerns = dict(collection.write_concern.document.items())
     combined_concerns.update(write_concerns)
     yield collection.with_options(write_concern=WriteConcern(**combined_concerns))
+
+
+@contextmanager
+def set_read_write_concern(collection, write_concerns, read_concern):
```

Comment:
I think its confusing that `write_concerns` is a dict and read_concern is a `ReadConcern`. Since write_concern is already using a dict, I would suggest to use a dict for ReadConcern too